Packs/SMIME_Messaging/Integrations/SMIME_Messaging/SMIME_Messaging.py

```diff
@@ -123,35 +407,77 @@ def verify(client: Client, args: Dict):
 
     """
     signed_message = demisto.getFilePath(args.get('signed_message'))
+    cert = args.get('public_key', 'instancePublicKey')
+    raw_output = argToBoolean(args.get('raw_output', 'false'))
+    tags = argToList(args.get('tag', ''))
 
-    x509 = X509.load_cert(client.public_key_file)
     sk = X509.X509_Stack()
-    sk.push(x509)
-    client.smime.set_x509_stack(sk)
-
     st = X509.X509_Store()
-    st.load_info(client.public_key_file)
+    if cert == 'instancePublicKey':
+        sk.push(X509.load_cert(client.public_key_file))
+        st.load_info(client.public_key_file)
+    elif cert:
+        public_key_file = NamedTemporaryFile(delete=False)
+        public_key_file.write(bytes(cert, 'utf-8'))
+        public_key_file.close()
+        sk.push(X509.load_cert(public_key_file.name))
+        st.load_info(public_key_file.name)
+        os.unlink(public_key_file.name)
```

Comment:
```suggestion
    elif cert:
        with NamedTemporaryFile(delete=False) as public_key_file:
            public_key_file.write(cert.encode('utf-8'))
            public_key_file_name = public_key_file.name

        sk.push(X509.load_cert(public_key_file_name))
        st.load_info(public_key_file_name)
        os.unlink(public_key_file_name)
```

with statement for creating and writing to the file, ensuring it is properly closed.